app/Http/Controllers/ApiController.php

```diff
@@ -160,68 +160,72 @@ public function cal($type)
                         ],
                     ] : null,
                 ],
+                'structured_data' => json_encode($this->getStructuredData($event)),
             ];
             $k++;
         }
 
         return response()->json($data);
     }
 
-    // Exposes information about one camp (by ID) for website integration
-    public function campInfo($camp_id)
+    // Construct event data as JSON-LD to please our Google overlords
+    // https://developers.google.com/search/docs/appearance/structured-data/event
+    private function getStructuredData(Event $event): ?array
     {
-        $camp = Event::find($camp_id);
-
-        if ($camp->type !== 'kamp') {
+        if (! in_array($event->type, ['kamp', 'online'], true)) {
             return null;
         }
```

Comment:
Ik denk dat null stringifyen niet zo chill checkt aan de frontend. Misschien is het makkelijker als de structured data property niet bestaat

Reply:
Hoe werkt dat met de schema definitie in typescript, krijg je dan geen errors daar?

Reply:
In typescript is het een string|undefined. In gatsby is het een `?String` dat gaat wel werken.